src/smalloc.cc

```diff
@@ -129,9 +141,13 @@ void CallbackInfo::WeakCallback(Isolate* isolate, Local<Object> object) {
     array_length *= array_size;
   }
   object->SetIndexedPropertiesToExternalArrayData(nullptr, array_type, 0);
-  int64_t change_in_bytes = -static_cast<int64_t>(array_length + sizeof(*this));
-  isolate->AdjustAmountOfExternalAllocatedMemory(change_in_bytes);
+  if (ownership_ == kInternal) {
+    int64_t change_in_bytes = -static_cast<int64_t>(array_length);
+    isolate->AdjustAmountOfExternalAllocatedMemory(change_in_bytes);
+  }
   callback_(static_cast<char*>(array_data), hint_);
+  isolate->AdjustAmountOfExternalAllocatedMemory(
+      -static_cast<int64_t>(sizeof(*this)));
```

Comment:
It's arguably better to call Isolate::AdjustAmountOfExternalAllocatedMemory() only once.  As is, the code may potentially trigger a GC twice.